promql: make lookback and matrix selections left-open and right-closed (#13904)
prometheus/prometheus: docs/querying/basics.md

select a range of samples back from the current instant. Syntactically, a [time
duration](#time-durations) is appended in square brackets (`[]`) at the end of
a vector selector to specify how far back in time values should be fetched for
each resulting range vector element. The range is a left-open and right-closed interval,
i.e. samples with timestamps coinciding with the left boundary of the range are

> excluded from the selection, while with the right boundary of the range are

Reviewer: ```suggestion
excluded from the selection, while samples coinciding with the right boundary of the range are
```

included in the selection.

> In this example, we select all the values we have recorded within the last 5 minutes,
> and not coinciding with exactly 5 minutes ago for all time series

Reviewer: Maybe let's word this in a more straightforward way:

```suggestion
In this example, we select all the values recorded less than 5m ago for all time series
```

cases like aggregation (`sum`, `avg`, and so on), where multiple aggregated
time series do not precisely align in time. Because of their independence,
Prometheus needs to assign a value at those timestamps for each relevant time

> series. It does so by taking the newest sample before this timestamp within the left-open lookback period,

Reviewer: I'm not sure if "left-open" will be understood in this context. How about the following:

```suggestion
series. It does so by taking the newest sample that is less than the lookback period ago.
```
